datacite-registered-doi-handler/src/main/java/no/unit/nva/datacite/handlers/RegisteredDoiEventHandler.java

```diff
@@ -0,0 +1,81 @@
+package no.unit.nva.datacite.handlers;
+
+import static org.zalando.problem.Status.GONE;
+import com.amazonaws.services.lambda.runtime.Context;
+import java.net.URI;
+import no.unit.nva.datacite.commons.DataCiteMetadataResolver;
+import no.unit.nva.datacite.commons.DoiUpdateRequestEvent;
+import no.unit.nva.datacite.commons.PublicationApiClientException;
+import no.unit.nva.doi.DoiClient;
+import no.unit.nva.doi.datacite.clients.DataCiteClientV2;
+import no.unit.nva.doi.datacite.clients.exception.ClientException;
+import no.unit.nva.doi.models.Doi;
+import no.unit.nva.events.handlers.DestinationsEventBridgeEventHandler;
+import no.unit.nva.events.models.AwsEventBridgeDetail;
+import no.unit.nva.events.models.AwsEventBridgeEvent;
+import nva.commons.core.JacocoGenerated;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RegisteredDoiEventHandler extends DestinationsEventBridgeEventHandler<DoiUpdateRequestEvent, Void> {
+    private static final Logger logger = LoggerFactory.getLogger(RegisteredDoiEventHandler.class);
+    private final DoiClient doiClient;
+    private final DataCiteMetadataResolver dataCiteMetadataResolver;
+    public static final String SHOULD_REMOVE_METADATA_LOG_MESSAGE =
+        "Publication with id {} is 410 Gone, assuming it is deleted and doi should have metadata removed";
+    private static final String RECEIVED_REQUEST_TO_MAKE_DOI_REGISTERED_LOG =
+        "Received request to unpublish doi (make registered) for DOI {}, publication id {} and customer id {}";
```

Comment:
`Will attempt to transition DOI {} to Registered DOI (for publication {} and customer {})` 

Do we get a matching message when this is successful?

Reply:
yup! SUCCESSFUL_DOI_REGISTERED